doc/configuration.md

```diff
@@ -2,7 +2,48 @@
 
 The Ignition configuration is a JSON document conforming to the following specification, with **_italicized_** entries being optional:
 
-* **ignitionVersion** (integer): the version number of the spec. Must be `1`.
+* **ignition** (object): metadata about the configuration itself.
+  * **version** (string): the semantic version number of the spec. Must be at least `2.0.0`.
+  * **_append_** (object): append one or more configs to the current config.
+    * **_configs_** (list of objects): a list of the configs to be appended.
+      * **source** (object): the source of the config.
+        * **uri** (string): the URI of the remote config.
+      * **_verification_** (object): options related to the verification of the config.
```

Comment:
I'm also opting for an overly verbose structure to as to match other similar options. For example, if you look at https://github.com/crawford/ignition/commit/da52c35b5ec828ffcd2b8a1518087c033f40f68c, you'll see `contents.source.uri` and `contents.source.immediate`. I'm using that same structure here, even though only `configs.source.uri` exists. This is just to help reduce the cognitive overhead of writing correct configs. Thooooough, if we are telling people that humans shouldn't write the configs, what does it matter?